### PR TITLE
allow sippy to wire against kube

### PR DESCRIPTION
**Pull request by deads2k**
Refactors the args to allow execution like `./sippy --server --local-data ../data-dir --dashboard=kube-master=sig-release-master-blocking,sig-release-master-informing=` that allows wiring against kubernetes (or non-openshift releases).

The `--release` flag still work and pipes to this new and specific mechanism.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -91,10 +93,45 @@ func (o *Options) Complete() error {
...
 }
 
+func (o *Options) ToTestGridDashboardCoordinates() []sippyserver.TestGridDashboardCoordinates {
+	dashboards := []sippyserver.TestGridDashboardCoordinates{}
+	for _, dashboard := range o.Dashboards {
+		tokens := strings.Split(dashboard, "=")
+		if len(tokens) != 3 {
+			// launch error
+			panic(fmt.Sprintf("must have three tokens: %q", dashboard))
+		}
```
Comment (abhinavdahiya) [inline main.go:110]: could we move this to validate function? making sure the Dashboard is always in the format we need?

Comment (deads2k) [reply main.go:110]: > could we move this to validate function? making sure the Dashboard is always in the format we need?

I would still keep this panic, but sure, adding it to validate is reasonable.

File: pkg/sippyserver/server.go [hunk 2]
```diff
@@ -69,63 +89,86 @@ func (s *Server) refresh(w http.ResponseWriter, req *http.Request) {
...
-		s.currTestReports[release].CurrentTwoDayReport,
-		s.currTestReports[release].PreviousWeekReport,
+		s.currTestReports[dashboard.ReportName].CurrentPeriodReport,
+		s.currTestReports[dashboard.ReportName].CurrentTwoDayReport,
+		s.currTestReports[dashboard.ReportName].PreviousWeekReport,
 		s.testReportGeneratorConfig.RawJobResultsAnalysisConfig.NumDays,
 		15)
 }
 
+func (s *Server) reportNameToDashboardCoordinates(release string) (TestGridDashboardCoordinates, bool) {
```
Comment (bparees) [inline pkg/sippyserver/server.go:118]:
```suggestion
func (s *Server) reportNameToDashboardCoordinates(reportName string) (TestGridDashboardCoordinates, bool) {
```

?

**Review by bparees — COMMENTED** [review]
nit and lgtm

File: main.go [hunk 3]
```diff
@@ -91,23 +93,70 @@ func (o *Options) Complete() error {
...
 	case "json":
 	default:
 		return fmt.Errorf("invalid output type: %s\n", o.Output)
 	}
 
+	for _, dashboard := range o.Dashboards {
+		tokens := strings.Split(dashboard, "=")
+		if len(tokens) != 3 {
+			// launch error
+			fmt.Errorf("must have three tokens: %q", dashboard)
```
Comment (bparees) [inline main.go:146]: shouldn't this be returned?

**Pull request merged**